src/Microsoft.AspNet.SignalR.JS/jquery.signalR.transports.common.js

```diff
@@ -672,15 +673,19 @@
         },
 
         verifyLastActive: function (connection) {
-            if (new Date().getTime() - connection._.lastActiveAt >= connection.reconnectWindow) {
-                var message = signalR._.format(signalR.resources.reconnectWindowTimeout, new Date(connection._.lastActiveAt), connection.reconnectWindow);
-                connection.log(message);
-                $(connection).triggerHandler(events.onError, [signalR._.error(message, /* source */ "TimeoutException")]);
-                connection.stop(/* async */ false, /* notifyServer */ false);
-                return false;
+            // If there is no keep alive configured, we cannot assume that timer callbacks will
+            // run frequently enough to keep lastActiveAt updated.
+            // https://github.com/SignalR/SignalR/issues/4536
+            if (!connection._.keepAliveData.activated ||
+                new Date().getTime() - connection._.lastActiveAt < connection.reconnectWindow) {
```

Comment:
What about using a "forward only" time like [`performance.now()`](https://developer.mozilla.org/docs/Web/API/Performance/now) which does not depend on the system clock which might change with timezones / NTP / manually?